lib/src/impl_js/impl_js.random.dart

```diff
@@ -16,8 +16,34 @@ part of 'impl_js.dart';
 
 void fillRandomBytes(TypedData destination) {
   try {
-    subtle.getRandomValues(destination);
+    if (destination is Uint8List) {
+      final newValues = subtle.getRandomValues(destination);
+      destination.setAll(0, newValues as Uint8List);
+    } else if (destination is Uint16List) {
+      final newValues = subtle.getRandomValues(destination);
+      destination.setAll(0, newValues as Uint16List);
+    } else if (destination is Uint32List) {
+      final newValues = subtle.getRandomValues(destination);
+      destination.setAll(0, newValues as Uint32List);
+    } else if (destination is Int8List) {
+      final newValues = subtle.getRandomValues(destination);
+      destination.setAll(0, newValues as Int8List);
+    } else if (destination is Int16List) {
+      final newValues = subtle.getRandomValues(destination);
+      destination.setAll(0, newValues as Int16List);
+    } else if (destination is Int32List) {
+      final newValues = subtle.getRandomValues(destination);
+      destination.setAll(0, newValues as Int32List);
+    } else {
+      throw UnsupportedError(
+        'Unsupported TypedData type: ${destination.runtimeType}',
+      );
+    }
   } on subtle.JSDomException catch (e) {
     throw _translateDomException(e);
+  } on UnsupportedError {
+    rethrow;
+  } on Error catch (e) {
+    throw _translateJavaScriptException(e);
```

Comment:
I'm not necessarily convinced this is a good idea.

We don't want to catch errors that we're not supposed to catch.

Hmm, I'm not sure exactly how we should handle this.

I'm guessing it means that the `_handleDomException` logic we have also won't work as intended.

I wonder if it's:
(a) Possible to get access to the JSObject that was thrown (somehow).
(b) Is there a bug for how error handling differs between js and wasm.

I'm betting that we don't have good enough tests in this package for all the exception cases, hence, also why a lot of tests didn't start failing when `_handleDomException` doesn't work as expected anymore.

Maybe, we need to put the error handling logic that differs into `crypto_suble.dart` such that `impl_js/` doesn't have to care whether it's running in dart2js or dart2wasm.

But I'm surprised that error handling differs between js and wasm, maybe we're doing something wrong :rofl:

Reply:
Thanks for referencing https://github.com/dart-lang/sdk/issues/55496

I'd suggest we look at `toString()`, because the only error we want to handle here is one that is `_JavaScriptError`. We might even want to make this code not run when it's compiled with dart2js (not sure if we can detect wasm vs js, though :rofl:)

Reply:
I would modify the code as follows 👍 
I wasn't expecting `Error` handling, so I was confused when implementing it. Thanks for the review.

```dart
} on Error catch (e) {
    final errorName = e.toString();
    if (errorName != 'JavaScriptError') {
        rethrow;
    }

    throw _translateJavaScriptException(e);
}
```

Reply:
I think this is a good idea, we probably have to do that in `_handleDomException` and in `getRandomValues` inside `lib/src/crypto_subtle.dart`.